test/models/ocelot/CalloutStanzaSpec.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.ocelot
+
+import org.scalatest._
+import play.api.libs.json._
+
+import scala.util.{Failure, Success, Try}
+
+class CalloutStanzaSpec extends FlatSpec {
+
+  val titleCalloutStanzaInputJson =
+      """|{
+         |    "type": "CalloutStanza",
+         |    "noteType": "Title",
+         |    "text": 0,
+         |    "next": ["1"],
+         |    "stack": false
+         |}""".stripMargin
+
+  val subTitleCalloutStanzaInputJson =
+    """|{
+       |    "type": "CalloutStanza",
+       |    "noteType": "Subtitle",
+       |    "text": 1,
+       |    "next": ["2"],
+       |    "stack": true
+       |}""".stripMargin
+
+  val missingPropertyCalloutStanzaInputJson =
+    """|{
+       |    "type": "CalloutStanza",
+       |    "noteType": "Title",
+       |    "text": 0,
+       |    "stack": false
+       |}""".stripMargin
+
+  val incorrectTypeCalloutStanzaInputJson =
+    """|{
+       |    "type": "CalloutStanza",
+       |    "noteType": "Title",
+       |    "text": 0,
+       |    "next": [1],
+       |    "stack": false
+       |}""".stripMargin
+
+  "read valid Callout stanza of type Title" should "Create a Callout stanza with a note type of Title" in {
+
+    val titleCalloutStanzaJson: JsValue = Json.parse( titleCalloutStanzaInputJson )
+
+    val titleCalloutStanza: CalloutStanza = titleCalloutStanzaJson.as[CalloutStanza]
+
+    assert( titleCalloutStanza == CalloutStanza( "Title", 0, Seq( "1"), false ) )
+
+  }
+
+  "read valid Callout stanza of type subtitle" should "Create a Callout stanza with a note type of Subtitle" in {
+
+    val subTitleCalloutStanzaJson : JsValue = Json.parse( subTitleCalloutStanzaInputJson )
+
+    val subTitleCalloutStanza: CalloutStanza = subTitleCalloutStanzaJson.as[CalloutStanza]
+
+    assert( subTitleCalloutStanza == CalloutStanza( "Subtitle", 1, Seq( "2"), true ) )
+  }
+
+  "read Callout stanza with missing property" should "Cause an exception to be raised" in {
```

Comment:
Story is about parsing valid JSON so this test was technically unnecessary. Though I happy to let it slide 😜